alf/algorithms/distributed_off_policy_algorithm.py

```diff
@@ -241,21 +239,27 @@ def receive_experience_data(replay_buffer: ReplayBuffer,
                 addr = 'tcp://' + ':'.join([unroller_ip, str(unroller_port)])
                 # Connect to an additional ROUTER
                 socket.connect(addr)
+        except queue.Empty:
+            pass
+
         if socket is not None:
             # Receive data from any router
             unroller_id, message = socket.recv_multipart()
-            if message == UnrollerMessage.EXP_SEG_END.encode():
+
+            buffer = io.BytesIO(message)
+            exp_params = torch.load(buffer, map_location='cpu')
+            # Use a temp buffer to store the received exps
+            if unroller_id not in unroller_exps_buffer:
+                unroller_exps_buffer[unroller_id] = []
+            unroller_exps_buffer[unroller_id].append(exp_params)
+
+            if int(exp_params.step_type) == StepType.LAST:
                 # Add the temp exp buffer to the replay buffer
-                for exp_params in unroller_exps_buffer[unroller_id]:
-                    replay_buffer.add_batch(exp_params, exp_params.env_id)
+                env_id = torch.zeros([1], dtype=torch.int32)
```

Comment:
device="cpu"

Reply:
Should change `exp_arams.env_id` instead, since exp_params.env_id will be stored in replay buffer and we dont' want inconsistency.
```python
exp_params.env_id.zero_()
```